### PR TITLE
feat: optimizing marketplace transactions

### PR DESCRIPTION
**Pull request by juliswer**

### CODE REVIEW

**Comment by AlefrankM** [comment]
Ay, your branch will have problems with mine

File: src/features/leda-nft/services/leda-nft.service.ts [hunk 1]
```diff
@@ -27,6 +27,14 @@ export default class LedaNftService implements INftService {
   public async approveForAll(address: string): Promise<void> {
     await this.contract?.setApprovalForAll(address, true);
   }
+
+  public async isApproveForAll(
+    ownerAddress: string,
+    marketPlaceAddress: string
+  ): Promise<Boolean | undefined> {
+    const result = await this.contract?.isApprovedForAll(ownerAddress, marketPlaceAddress);
```
Comment (AlefrankM) [inline src/features/leda-nft/services/leda-nft.service.ts:35]: You will not use this result, so the await is unnecesary

File: src/features/marketplace/process/commands/list-item/list-item-command.ts [hunk 2]
```diff
@@ -24,7 +24,8 @@ export default class MakeItemNftCommand implements ICommand<MarketplaceState> {
       const transaction = await this.marketplaceService.listItem(
         state.collectionAddress,
         state.tokenId,
-        wei
+        wei,
```
Comment (mmiranda-stackit) [inline src/features/marketplace/process/commands/list-item/list-item-command.ts:27]: If you work on a file and it has no tests, you should add them

File: src/features/marketplace/services/marketplace.service.ts [hunk 3]
```diff
@@ -28,9 +28,14 @@ export default class MarketplaceService implements IMarketplaceService {
...
-    price: string
+    price: string,
+    ownerAddress: string
   ): Promise<ContractTransaction | undefined> {
-    await this.ledaNftService.approveForAll(marketplaceAddress.address);
+    const approvedNft = await this.ledaNftService.isApproveForAll(
+      ownerAddress,
+      marketplaceAddress.address
+    );
+    if (!approvedNft) await this.ledaNftService.approveForAll(marketplaceAddress.address);
```
Comment (mmiranda-stackit) [inline src/features/marketplace/services/marketplace.service.ts:38]: nice

**Review by mmiranda-stackit — APPROVED** [review]
approved with comments

**Pull request merged**